src/mrsimulator/method.py

```diff
@@ -0,0 +1,608 @@
+    #                     # print('D_segment: ', D_segment)
+
+    #             segments.append(np.asarray(P_segment))  # append the intersection
+    #     return cartesian_product(*segments)
+
+    def apodize(self, **kwargs):
+        """Returns the result of passing the selected apodization function .
+
+        Args:
+            csdm: simulation object
+
+        Returns:
+            A Numpy array
+        """
+
+        csdm = self.simulation
+        for dim in csdm.dimensions:
+            dim.to("Hz", "nmr_frequency_ratio")
+        apo = self.post_simulation.apodization
+
+        sum_ = 0
+
+        for item in apo:
+            sum_ += item.apodize(csdm)
+
+        for dim in csdm.dimensions:
+            dim.to("ppm", "nmr_frequency_ratio")
+
+        return self.post_simulation.scale * sum_
+
+
+def cartesian_product(*arrays):
+    la = len(arrays)
+    dtype = np.result_type(*arrays)
+    arr = np.empty([len(a) for a in arrays] + [la], dtype=dtype)
+    for i, a in enumerate(np.ix_(*arrays)):
+        arr[..., i] = a
+    return arr.reshape(-1, la)
+
+
+def P_symmetry_indexes(transitions, list_of_P):
+    P = transitions[:, 1, :] - transitions[:, 0, :]
+    n = P.shape[1]
+    return reduce(
+        np.union1d,
+        [
+            reduce(np.intersect1d, [np.where(P[:, i] == search[i]) for i in range(n)])
+            for search in list_of_P
+        ],
+        np.asarray([], dtype=np.int64),
+    )
+
+
+def D_symmetry_indexes(transitions, list_of_D):
+    D = transitions[:, 1, :] ** 2 - transitions[:, 0, :] ** 2
+    return reduce(
+        np.union1d,
+        [
+            reduce(
+                np.intersect1d,
+                [np.where(D[:, i] == search[i]) for i in range(D.shape[1])],
+            )
+            for search in list_of_D
+        ],
+        np.asarray([], dtype=np.int64),
+    )
+
+
+def get_iso_dict(channel, isotope):
+    """
+        Parse the isotopomer sites to determine indices of each isotope that
+        is part of the method channel.
+
+        Args:
+            channel: List object
+            isotope: List object
+
+    """
+    iso_dict = {}
+
+    # determine channels for P
+    for i, item in enumerate(isotope):
+        if item in channel and item not in iso_dict:
+            iso_dict[item] = [i]
+        elif item in iso_dict:
+            iso_dict[item].append(i)
+
+    return iso_dict
+
+
+def query_permutations(query, isotope, channel, transition_symmetry="P"):
+    """
+        Determines the transition symmetries that are involved in a given
+        transition query.
+
+        Args:
+            query: Dict object
+            channel: List object
+            isotope: List object
+            transition_symmetry: str object. Derived from a transition query
+
+    """
+
+    P_permutated = []
+    iso_dict = get_iso_dict(channel=channel, isotope=isotope)
+    query_short = query[transition_symmetry]
```

Comment:
Is there a reason for the argument `transition_symmetry="P"` in the `query_permutations()` function? Can you not query for `P` from the `query` argument itself as,
```
if "P" in query.keys():
  query_short = query["P"]